### DISCUSSION TITLE
Akeeba Backup JSON API Error - Akeeba Panopticon

### DISCUSSION THREAD

**Discussion by aboarken** [akeeba/panopticon#244] [Q&A (Community Support)] [answered]
Hello again Nicholas,

I am getting this error on all my added websites, should I have to do something from my side?

 Akeeba Backup JSON API Error
There was an error connecting to your site's Akeeba Backup JSON API.
Error Type: Akeeba\BackupJsonApi\Exception\InvalidSecretWord
Error Code: 42
Error Message: Authentication error (invalid Secret Word). Please check the secret word, make sure it doesn't have any whitespace you missed. Clear any site or external caches, making sure Akeeba Backup's URL isn't cached.

Please check attached screenshot:

![Screenshot_1](https://github.com/akeeba/panopticon/assets/18500882/ace8cad4-514b-468c-a74b-7ee51fed1e7e)

**Comment by Martin511** [comment]
Hello Ahmad,
did you enable front-end and remote backup in the Akeeba Backup settings in Joomla?
And did you set a secret word at all?

**Comment by aboarken** [comment]
> Hello Ahmad, did you enable front-end and remote backup in the Akeeba Backup settings in Joomla? And did you set a secret word at all?

Hello @Martin511, can you please tell me which settings? Or where I can find them? How to set the Secret Word? Thank you.

**Comment by Martin511** [comment]
Hello Ahmad,
it is explained in teh first minute of this video: https://www.akeeba.com/videos/1212-akeeba-backup/1616-abtc02-automating-backup.html

**Comment by aboarken** [comment]
Thank you for your support @Martin511. I have found that this option "Enable JSON API (remote backup)" is enabled but this legacy option "Enable Legacy Front-end Backup API (remote CRON jobs)" is disabled as shown in the below picture. The secret word was already available but still the issue exists even if I enable the legacy option. Any suggestions? Did it work for you?

![Screenshot_1](https://github.com/akeeba/panopticon/assets/18500882/02bbe762-07db-4ca5-a27b-24d0dd29ccfd)

**Comment by nikosdion** [comment]
Try a more complex secret word. If Akeeba Backup thinks that the Secret Word is not complex / random enough it will reject it and won't allow the JSON API to operate. This is a security safeguard. You wouldn't believe just how many people would put `PassW0rd` or `spike123` in there…

Here is how to do this.

Go to https://www.random.org/passwords/?num=2&len=32&format=html&rnd=new to get two new 32-character passwords. For example, let's say it gives you these:

* `498K97UKb375ZAYSfrwA9XayLVjhKmdU`
* `rCguwymGQTVsGfRhetzK6RDEWuSUJZza`

Put them together into one long password. In the example above that would be `498K97UKb375ZAYSfrwA9XayLVjhKmdUrCguwymGQTVsGfRhetzK6RDEWuSUJZza`

Put this long password into the Secret Word configuration option.

**Comment by Martin511** [comment]
Because I'm a lazy dog, I always do it like this:
Activate front-end and remote backup.
Leave the "Secret Word" field empty.
![Screenshot 2023-10-30 184106](https://github.com/akeeba/panopticon/assets/56762649/a9c9efee-39c5-405b-b35f-cecfdf11356b)

Click on Save & Close.
Afterwards you get a message that the secret word has not been set and the system offers you a suggestion.
Click on "Apply the suggested Secret Word".
![Screenshot 2023-10-30 184145](https://github.com/akeeba/panopticon/assets/56762649/48231070-f87b-4967-b160-7e5619bee5b6)

Done.

**Comment by nikosdion** [comment]
I have seen exactly ONE (1) server in the last 8 years where the auto-generated Secret Word wasn't complex enough (I have no idea what's broken on that server). I suggested the hard way in the unlikely but not impossible case there are _two_ servers like that 😆

**Comment by aboarken** [comment]
Hello @Martin511, thank you for your reply. I have already applied all your suggestions but unfortunately the issue is not yet solved.

**Comment by nikosdion** [comment]
Can you please try using the secret word `Ldc9AtvtG6SUBshCTwZ7EsaevG4qhuMZwJVLAKxnF2d6wW9JYNBgSJtJvQhe4q2b` and tell us if it works?

**Comment by aboarken** [comment]
Hello @nikosdion, I have used the mentioned secret word but it didn't work.

**Comment by nikosdion** [comment]
Does it still give you the error InvalidSecretWord, or something different?

**Comment by aboarken** [comment]
Its giving me same error @nikosdion.

**Comment by nikosdion** [comment]
Use the contact form on our business site to give me the site's URL and the Panopticon token. I want to see what information it receives from the API now that I know what the secret word is.

**Comment by komir2004** [comment]
Hi I have similar problem but only with J3 pages, cant re–establish the link with Akeeba Backup Professional.
![Akeeba-Panopticon-3](https://github.com/akeeba/panopticon/assets/17553018/990e9e47-545c-4c70-84f5-c56c5404449b)
With debug enabled I don't get any additional information.

On J4 pages Akeeba Backup is working.

I am using Akeeba Backup Professional 8.3.3, Joomla! 3.10.12, PHP Version 8.1.24, Database Version 8.0.35
Enable JSON API (remote backup) is enabled with secret word suggested here (https://github.com/akeeba/panopticon/discussions/244#discussioncomment-7491891)

**Comment by nikosdion** [comment, marked as answer]
Everyone who has a problem connecting their sites, please do read and follow the troubleshooting information in https://github.com/akeeba/panopticon/wiki/Managing-Sites#troubleshooting-the-connection  This is the sum total of problems I have seen to date, and suggestions for resolving them.
